lib/faktory_worker/worker.ex

```diff
@@ -140,21 +152,21 @@ defmodule FaktoryWorker.Worker do
   end
 
   defp handle_fetch_response({{:ok, job}, conn}, state) do
-    job_supervisor =
-      state.worker_config
-      |> faktory_name()
-      |> FaktoryWorker.JobSupervisor.format_supervisor_name()
+    job_supervisor = job_supervisor_name(state)
 
     job_ref =
       Task.Supervisor.async_nolink(
         job_supervisor,
         state.worker_module,
         :perform,
         job["args"],
-        # make this whatever is set for the 'reserve_for' minus 20 seconds
-        shutdown: :infinity
+        shutdown: :brutal_kill
       )
 
+    reserve_for = Map.get(job, "reserve_for", @faktory_default_reserve_for)
+
+    Process.send_after(self(), :job_timeout, reserve_for - 20)
```

Comment:
I think it would be useful to have a comment here describing why we're using 20 seconds less than the reserve for time.